forbidigo/forbidigo.go

```diff
@@ -97,19 +100,48 @@ type visitor struct {
 	linter   *Linter
 	comments []*ast.CommentGroup
 
-	fset   *token.FileSet
-	issues []Issue
+	runConfig RunConfig
+	issues    []Issue
 }
 
+// Deprecated: Run was the original entrypoint before RunWithConfig was introduced to support
+// additional match patterns that need additional information.
 func (l *Linter) Run(fset *token.FileSet, nodes ...ast.Node) ([]Issue, error) {
-	var issues []Issue 
+	return l.RunWithConfig(RunConfig{Fset: fset}, nodes...)
+}
+
+// RunConfig provides information that the linter needs for different kinds
+// of match patterns. Ideally, all fields should get set. More fields may get
+// added in the future as needed.
+type RunConfig struct {
+	// FSet is required.
+	Fset *token.FileSet
+
+	// TypesInfo is needed for expanding source code expressions.
+	// Nil disables that step, i.e. patterns match the literal source code.
+	TypesInfo *types.Info
+
+	// DebugLog is used to print debug messages. May be nil.
+	DebugLog func(format string, args ...interface{})
+}
+
+// Patterns returns the parsed patterns.
+func (l *Linter) Patterns() []*Pattern {
```

Comment:
Are we using this for anything still or can we remove it?

Reply:
We can remove it.